src/parser/src/main/java/io/github/syst3ms/skriptparser/sections/SecWhile.java

```diff
@@ -58,7 +58,9 @@ public Statement setNext(@Nullable Statement next) {
     }
 
     /**
-     * @see SecLoop#getActualNext()
+     * This method exists because SecWhile actually sets itself as its next element with {@link #getNext()}.
+     * This way it has full control over when to stop iterating.
+     * @return the element that is actually after this SecWhile
      */
     @Nullable
     public Statement getActualNext() {
```

Comment:
This needs to be an Optional, but I take care of that in my PR, so you'll be able to add it easily soon.